bot/bot.go

```diff
@@ -32,42 +32,95 @@ type Bot interface {
 	Channels() []string                                       // Channels returns list of channels which bot has joined.
 }
 
-// NewBot creates a new Bot implementation with a set of parameters.
-func NewBot(host string, ssl bool, nick, ident string, channels []string) Bot {
-	return NewBotWithProxy(host, ssl, nick, ident, channels, "")
-}
-
-// NewBotWithProxy creates a new Bot implementation with a set of parameters and a proxy.
-func NewBotWithProxy(host string, ssl bool, nick, ident string, channels []string, proxy string) Bot {
+// NewBotWithProxyPassword creates a new Bot implementation with required parameters and options.
+func NewBotOptions(host string, nick string, options ...func(*BotImpl)) (Bot, error) {
```

Comment:
How about we make everything an option for consistency, especially for multiple hosts, and argument order insensitivity?